src/test/java/io/aiven/kafka/connect/http/sender/ApiKeyHttpSenderTest.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2021 Aiven Oy and http-connector-for-apache-kafka project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.mockito.stubbing.Answer;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
+class ApiKeyHttpSenderTest {
+
+    @Mock
+    HttpClient mockedHttpClient;
+
+    final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Test
+    void buildAccessTokenAuthHeaderForDefaultSettings(@Mock final HttpResponse<String> accessTokenResponse)
+            throws IOException, InterruptedException {
+        final var config = defaultConfig();
+
+        final var httpSend =
+                new ApiKeyHttpSender(
+                        new HttpSinkConfig(config),
+                        mockedHttpClient
+                );
+
+        final var requestCaptor = ArgumentCaptor.forClass(HttpRequest.class);
+
+        final var accessTokenJson = Map.of(
+                "access_token", "bla-bla-bla"
+        );
+
+        when(accessTokenResponse.statusCode()).thenReturn(200);
+        when(accessTokenResponse.body()).thenReturn(objectMapper.writeValueAsString(accessTokenJson));
+        when(mockedHttpClient.<String>send(requestCaptor.capture(), any())).thenReturn(accessTokenResponse);
```

Comment:
this test is not very useful, you don't check that the you are calling with the ApiKey here. Using any() will answer a mocked accessToken to any request.